lib/gds_api/publishing_api_v2.rb

```diff
@@ -1,497 +1,9 @@
-      while next_link
-        yielder.yield begin
-          response = get_json(next_link)
-        end
-        next_link_info = response["links"].select { |link| link["rel"] == "next" }.first
-        next_link = next_link_info && next_link_info["href"]
-      end
-    end
-  end
-
-  # Returns a mapping of content_ids => links hashes
-  #
-  # @param content_ids [Array]
-  #
-  # @return [Hash] a mapping of content_id => links
-  #
-  # @example
-  #
-  #   publishing_api.get_links_for_content_ids([
-  #     "e1067450-7d13-45ff-ada4-5e3dd4025fb7",
-  #     "72ed754c-4c82-415f-914a-ab6760454cb4"
-  #   ])
-  #
-  #   #=> {
-  #     "e1067450-7d13-45ff-ada4-5e3dd4025fb7" => {
-  #       links: {
-  #         taxons: ["13bba81c-b2b1-4b13-a3de-b24748977198"]},
-  #         ... (and more attributes)
-  #       version: 10},
-  #     "72ed754c-4c82-415f-914a-ab6760454cb4" => { ..etc }
-  #   }
-  #
-  def get_links_for_content_ids(content_ids)
-    post_json("#{endpoint}/v2/links/by-content-id", content_ids: content_ids).to_hash
-  end
-
-  # Reserves a path for a publishing application
-  #
-  # Returns success or failure only.
-  #
-  # @param payload [Hash]
-  # @option payload [Hash] publishing_app The publishing application, like `content-tagger`
-  #
-  # @see https://docs.publishing.service.gov.uk/apis/publishing-api/api.html#put-pathsbase_path
-  def put_path(base_path, payload)
-    url = "#{endpoint}/paths#{base_path}"
-    put_json(url, payload)
-  end
-
-private
-
-  def content_url(content_id, params = {})
-    validate_content_id(content_id)
-    query = query_string(params)
-    "#{endpoint}/v2/content/#{content_id}#{query}"
-  end
-
-  def links_url(content_id)
-    validate_content_id(content_id)
-    "#{endpoint}/v2/links/#{content_id}"
-  end
-
-  def links_changes_url(params = {})
-    query = query_string(params)
-    "#{endpoint}/v2/links/changes#{query}"
-  end
-
-  def publish_url(content_id)
-    validate_content_id(content_id)
-    "#{endpoint}/v2/content/#{content_id}/publish"
-  end
-
-  def republish_url(content_id)
-    validate_content_id(content_id)
-    "#{endpoint}/v2/content/#{content_id}/republish"
-  end
-
-  def unpublish_url(content_id)
-    validate_content_id(content_id)
-    "#{endpoint}/v2/content/#{content_id}/unpublish"
-  end
-
-  def discard_url(content_id)
-    validate_content_id(content_id)
-    "#{endpoint}/v2/content/#{content_id}/discard-draft"
-  end
-
-  def get_editions_url(params)
-    query = query_string(params)
-    "#{endpoint}/v2/editions#{query}"
-  end
-
-  def merge_optional_keys(params, options, optional_keys)
-    optional_keys.each_with_object(params) do |optional_key, hash|
-      hash.merge!(optional_key => options[optional_key]) if options[optional_key]
-    end
-  end
-
-  def validate_content_id(content_id)
-    raise ArgumentError, "content_id cannot be nil" unless content_id
-  end
-end
+class GdsApi::PublishingApiV2 < GdsApi::PublishingApi; end
```

Comment:
There's probably a deprecation warning opportunity in the initialize method for this class: https://github.com/alphagov/gds-api-adapters/blob/5d732cc35ffc7441da6c7cfb7ac3fdb5d56464dc/lib/gds_api/rummager.rb